client/src/components/ExpandedRowComponent.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {MouseEvent} from 'react';
+import type {CodeReview} from '../lib/types';
+import {format} from 'date-fns';
+import {logError} from '../services/log';
+import {useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+export const useExpandedRowComponent = ({onUpdate, user}: {onUpdate: CallableFunction; user?: {displayName: string}}) => {
+    const ExpandedRowComponent = ({data}: {data: CodeReview}) => {
```

Comment:
This feels like a code smell to declare this component inside the hook and then have the hook's arguments in the component's scope. I would declare the component separately and pass `user` and `onUpdate` as props.

Update after reading through the rest: This isn't really a hook, it's a component that's hiding some additional props. I would:
- just call this component `ExpandedRow`, component in the name is redundant
- remove the nested component and make `data` one of ExpandedRow's props
- use PropTypes to require onUpdate, user and possibly data so the component will not render if they do not exist

Reply:
It's a callback function that return a component.  So it's tricky since the component itself rely on the input user that interact inside the table row that control by the table component.  Because the table component only know about the row record and doesn't know anything about the user data.  Since we can passed in a custom prop into the table component to pass to expanRowComponent, this is the only way to do it outside of the ReviewList component as a separate helper hook.

Reply:
Good point, I didn't think of that.

Reply:
Fixed in commit 2e58905276fb204dab0ef77f3de5aa028c7ea34e